R-package/R/utils.R

```diff
@@ -1,13 +1,13 @@
 lgb.is.Booster <- function(x) {
-  return(all(c("R6", "lgb.Booster") %in% class(x)))
+  return(all(inherits(c("R6", "lgb.Booster"), class(x))))
```

Comment:
```suggestion
  return(inherits(x, c("R6", "lgb.Booster")))
```

These changes are not correct. `inherits()` accepts an object followed by a character vector of class names. You are passing a vector of class names followed by another vector of class names.

Please fix this and the other `utils.R` changes. You can run `?inherits` in an R session to learn how that function works.

Reply:
Thanks for the review! I have updated PR description. I wonder if this lint is a false positive because we are using `all`. For example
``` R
> x <- 10
> class(x)
[1] "numeric"
> c("character", "numeric") %in% class(x)
[1] FALSE  TRUE
> all(c("character", "numeric") %in% class(x))
[1] FALSE
> inherits(x, c("character", "numeric"))
[1] TRUE
>
```
`all` while checking yields `FALSE`, `inherits` - `TRUE`. Could be the case when `x` inherits from `R6` but not from `lgb.Booster` or vise-versa. Is that a potential false positive or am I missing something?

Reply:
Ah thanks for that! I did not realize that `inherits()` performs that way with multiple classes. But I see that now in `?inherits`.

> inherits indicates whether its first argument inherits from **any** of the classes specified in the what argument

Reading the docs in `?lintr::class_equals_linter`, I think it was intended to only catch the opposite type of condition...code like this:

```r
x <- as.POSIXct("2021-01-01")
class(x)
# [1] "POSIXct" "POSIXt" 

# this results in a warning:
# "the condition has length > 1 and only the first element will be used"
if (class(x) == "numeric") {
    print("hello")
}
```

I still think this linter is useful for such cases. But I agree with you that these particular lines in `utils.R` are false positives.

Could you please revert changes in `utils.R`, and add comments like `  # nolint: class_equals_linter` (note the two leading spaces) on each offending line?

Reply:
Latest commit is up with updated branch!

Reply:
Shall I update my PR description mentioning/copying our conversation into it? I believe resolved convo often are initially displayed collapsed, so someone looking for why these warning were suppressed might not see this.

Reply:
Thanks! No, no need to do that. Search engines are still able to find collapsed conversations on GitHub, and anyone looking at the project's source code can walk back through the blame view ([docs link](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#viewing-the-line-by-line-revision-history-for-a-file)) to find this PR.